### PR TITLE
🐛 Check existence of MachineControlPlaneLabelName label instead of value

### LINKED ISSUE #1697: util.IsControlPlaneMachine(machine *clusterv1.Machine) is true if the label value is "false"
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]
As far as I understand it, for a control plane `machine` we set a label `cluster.x-k8s.io/control-plane: "true"`, I would presume that for a worker node we would set the label as false, or at least could do?

However the check for this label to be determined true is that the label isn't empty, so  `cluster.x-k8s.io/control-plane: "false"` is actually `true`. 

**What did you expect to happen:**
`false` to mean `false`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version: 0.2.6
- Minikube/KIND version: N/A
- Kubernetes version: (use `kubectl version`): 1.14.x
- OS (e.g. from `/etc/os-release`):

/kind bug

### PR DESCRIPTION
**Pull request by tahsinrahman**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1697

### CODE REVIEW

**Comment by ncdc** [comment]
Should we add this to the v1a2 to v1a3 doc, that we now accept the empty string for this label whereas before we did not?

File: docs/book/src/providers/v1alpha2-to-v1alpha3.md [hunk 1]
```diff
@@ -40,3 +40,7 @@
 ## The `util/restmapper` package has been removed
 
 - Controller runtime has native support for a [DynamicRESTMapper](https://github.com/kubernetes-sigs/controller-runtime/pull/554/files), which is used by default when creating a new Manager.
+
+## The `util.IsControlPlaneMachine` is now accepts empty value of `MachineControlPlaneLabelName` label
```
Comment (tahsinrahman) [inline docs/book/src/providers/v1alpha2-to-v1alpha3.md]: @ncdc updated docs here, ptal :)

**Comment by tahsinrahman** [comment]
@detiber rebased and resolved conflicts

File: docs/book/src/providers/v1alpha2-to-v1alpha3.md (hunk 1 above)
Comment (chuckha) [reply docs/book/src/providers/v1alpha2-to-v1alpha3.md]: I think the thing to point out in this document is that if the label is set the machine is determined to be a control plane. It does more than accept empty values, it accepts *any* value for the label.

File: bootstrap/kubeadm/controllers/kubeadmconfig_controller.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 	if !cluster.Status.ControlPlaneInitialized {
 		// if it's NOT a control plane machine, requeue
 		if !util.IsControlPlaneMachine(machine) {
-			log.Info(fmt.Sprintf("Machine is not a control plane. If it should be a control plane, add `%s: true` as a label to the Machine", clusterv1.MachineControlPlaneLabelName))
+			log.Info(fmt.Sprintf("Machine is not a control plane. If it should be a control plane, add an empty label `%s: \"\"` to the Machine", clusterv1.MachineControlPlaneLabelName))
```
Comment (detiber) [inline bootstrap/kubeadm/controllers/kubeadmconfig_controller.go]:
```suggestion
			log.Info(fmt.Sprintf("Machine is not a control plane. If it should be a control plane, add the label `%s: \"\"` to the Machine", clusterv1.MachineControlPlaneLabelName))
```

File: docs/book/src/providers/v1alpha2-to-v1alpha3.md [hunk 3]
```diff
@@ -44,3 +44,7 @@
 ## Generated kubeconfig admin username changed from `kubernetes-admin` to `<cluster-name>-admin`
 
 - The kubeconfig secret shipped with Cluster API now uses the cluster name as prefix to the `username` field.
+
+## The `util.IsControlPlaneMachine` is now accepts empty value of `MachineControlPlaneLabelName` label
```
Comment (detiber) [inline docs/book/src/providers/v1alpha2-to-v1alpha3.md]:
```suggestion
## The `util.IsControlPlaneMachine` is now accepts any value of `MachineControlPlaneLabelName` label
```

**Comment by tahsinrahman** [comment]
thanks @chuckha @detiber, updated!

File: docs/book/src/providers/v1alpha2-to-v1alpha3.md [hunk 4]
```diff
@@ -44,3 +44,7 @@
 ## Generated kubeconfig admin username changed from `kubernetes-admin` to `<cluster-name>-admin`
 
 - The kubeconfig secret shipped with Cluster API now uses the cluster name as prefix to the `username` field.
+
+## The `util.IsControlPlaneMachine` is now accepts any value of `MachineControlPlaneLabelName` label
```
Comment (vincepri) [inline docs/book/src/providers/v1alpha2-to-v1alpha3.md]:
```suggestion
## A Machine is now considered a control plane if it has `cluster.x-k8s.io/control-plane` set, regardless of value
```

File: docs/book/src/providers/v1alpha2-to-v1alpha3.md [hunk 5]
```diff
@@ -44,3 +44,7 @@
...
+
+- `util.IsControlPlaneMachine` doesn't require to set a value of `MachineControlPlaneLabelName` label, it only requires the existence of this label
```
Comment (vincepri) [inline docs/book/src/providers/v1alpha2-to-v1alpha3.md]:
```suggestion
- Previously examples and tests were setting/checking for the label to be set to `true`.
- The function `util.IsControlPlaneMachine` was previously checking for any value other than empty string, while now we only check if the associated label exists.
```

**Pull request merged**